static/app/views/performance/landing/widgets/widgets/performanceScoreListWidget.tsx

```diff
@@ -0,0 +1,282 @@
+import {transformDiscoverToList} from '../transforms/transformDiscoverToList';
+import {transformEventsRequestToStackedArea} from '../transforms/transformEventsToStackedBars';
+import {PerformanceWidgetProps, WidgetDataResult} from '../types';
+
+type DataType = {
+  chart: WidgetDataResult & ReturnType<typeof transformEventsRequestToStackedArea>;
+  list: WidgetDataResult & ReturnType<typeof transformDiscoverToList>;
+};
+
+export function PerformanceScoreListWidget(props: PerformanceWidgetProps) {
+  const location = useLocation();
+  const [selectedListIndex, setSelectListIndex] = useState<number>(0);
+  const {ContainerActions, organization, InteractiveTitle} = props;
+  const theme = useTheme();
+  const shouldUseStoredScores = useStoredScoresSetting();
+
+  // TODO Abdullah Khan: Create a new widget type/file for Best Page Opportunity
+  // Web vitals widget. Code path is very different from generic stacked area chart widget.
+  const {data: projectData, isLoading: isProjectWebVitalDataLoading} =
+    useProjectRawWebVitalsQuery();
+  const {data: projectScoresData, isLoading: isProjectScoresLoading} =
+    useProjectWebVitalsScoresQuery({
+      enabled: shouldUseStoredScores,
+    });
+
+  const projectScore = shouldUseStoredScores
+    ? calculatePerformanceScoreFromStoredTableDataRow(projectScoresData?.data?.[0])
+    : calculatePerformanceScoreFromTableDataRow(projectData?.data?.[0]);
+
+  const {data: transactionWebVitals, isLoading: isTransactionWebVitalsQueryLoading} =
+    useTransactionWebVitalsQuery({limit: 4});
+
+  const {data: timeseriesData, isLoading: isTimeseriesQueryLoading} =
+    useProjectWebVitalsTimeseriesQuery({});
+
+  const assembleAccordionItems = provided =>
+    getHeaders(provided).map(header => ({header, content: getAreaChart(provided)}));
+
+  const getAreaChart = _ =>
+    function () {
+      const segmentColors = theme.charts.getColorPalette(3);
+      return (
+        <Chart
+          stacked
+          height={props.chartHeight}
+          data={formatTimeSeriesResultsToChartData(
+            timeseriesData,
+            segmentColors,
+            !shouldUseStoredScores
+          )}
+          disableXAxis
+          loading={false}
+          utc={false}
+          grid={{
+            left: 5,
+            right: 5,
+            top: 5,
+            bottom: 0,
+          }}
+          dataMax={100}
+          chartColors={segmentColors}
+        />
+      );
+    };
+
+  const getHeaders = _ =>
+    transactionWebVitals.map(
+      listItem =>
+        function () {
+          const transaction = (listItem.transaction as string | undefined) ?? '';
+          const count = projectData?.data?.[0]?.['count()'] as number;
+          const scoreCount = projectScoresData?.data?.[0]?.[
+            'count_scores(measurements.score.total)'
+          ] as number;
+          const opportunity = shouldUseStoredScores
+            ? ((listItem.opportunity ?? 0) * 100) / scoreCount
```

Comment:
What happens to this score when your `as number` scoreCount above comes back as undefined or 0?

Reply:
So we shouldn't be able to enter this block of code since no rows would be returned anyways to iterate on, but I've added a check to just return an opportunity score of 0 for now.